app/log/invocation_log.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { invocation } from "../../proto/invocation_ts_proto";
+import { eventlog } from "../../proto/eventlog_ts_proto";
+import rpcService from "../service/rpc_service";
+import errorService from "../errors/error_service";
+import { BuildBuddyError } from "../util/errors";
+import Log from "./log";
+import FAKE_FETCHER from "./fake_fetcher";
+
+export type InvocationLogProps = {
+  invocationId: string;
+};
+
+type State = {
+  lines: string[] | null;
+  previousChunkId?: string;
+  nextChunkId?: string;
+};
+
+export default class InvocationLog extends React.Component<InvocationLogProps, State> {
```

Comment:
I know this is a WIP - but do we really need to create our own log component? There's just a lot of stuff we'd have to implement to replace LazyLog. Can we not just build a string by concatenating chunks and feed it to LazyLog with `text=` ?

Reply:
If we prepend chunks like `this.setState({ value: fetchedChunk + this.state.value })`, the scroll position gets messed up, because `react-lazylog` just re-renders. When `follow={true}` it will just scroll to the end of the logs whenever it re-renders.  So we have to somehow restore the scroll position. If we try to save the scroll position (as it was before prepending) and then restore it after the logs are re-rendered using something like `logElement.scrollTop = ...`, then react-lazylog spazzes out. I haven't dug much into it yet, but presumably it's because it has its own logic for scrolling (particularly, `react-virtualized` listens for scrolls and then updates which elements are rendered).

I'll look into it a bit more -- my gut feeling so far though is that we're going to have to wrestle pretty hard to get it to work.

I understand that react-lazylog solves some problems for us (virtualizing the list and also converting ANSI sequences to HTML) but I think the list virtualization piece is not a major benefit (especially since it causes a lot of problems, e.g. Ctrl+F doesn't work and it doesn't allow text wrapping which is very unfriendly for Bazel logs) and we can reuse the ANSI parsing libraries from react-lazylog to avoid having to implement it ourselves.